### PR TITLE
Fixed it so you don't have to go through all the names if it can't be…

### PR DESCRIPTION
**Pull request by bathmatt**
… at this point

@trilinos/<teamName>

### CODE REVIEW

File: packages/teuchos/comm/src/Teuchos_StackedTimer.cpp [hunk 1]
```diff
@@ -56,6 +56,12 @@ StackedTimer::LevelTimer::findBaseTimer(const std::string &name) const {
 BaseTimer::TimeInfo
 StackedTimer::LevelTimer::findTimer(const std::string &name, bool& found) {
   BaseTimer::TimeInfo t;
+  auto full_name = get_full_name();
+  auto level=0;
```
Comment (rppawlo) [inline packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: level looks to be unused.

Comment (bathmatt) [reply packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: you passed the test, very good...  (it was used before I realized my code could be much simpler)

File: packages/teuchos/comm/src/Teuchos_StackedTimer.cpp [hunk 2]
```diff
@@ -56,6 +56,11 @@ StackedTimer::LevelTimer::findBaseTimer(const std::string &name) const {
 BaseTimer::TimeInfo
 StackedTimer::LevelTimer::findTimer(const std::string &name, bool& found) {
   BaseTimer::TimeInfo t;
+  auto full_name = get_full_name();
+  if (full_name.size() > name.size())
```
Comment (mhoemmen) [inline packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: I think this is fine in context of the `strncmp` call below.  On the other hand, now you're doing two string comparisons.  I would have written the code like this:
```c++
const std::string full_name = get_full_name ();
if (full_name != name) {
  return t;
}
t = BaseTimer::TimeInfo(this);
found = true;
```

**Comment by mhoemmen** [comment]
Does this fix #6478 or is it just a part?

**Pull request merged**